src/main_loop.rs

```diff
@@ -90,30 +77,54 @@ pub(crate) struct MainLoop {
 }
 
 impl MainLoop {
-    async fn get_session(&mut self) -> Result<Session, SessionError> {
+    async fn get_connection(&mut self) -> Result<(Spirc, impl Future<Output = ()>), Error> {
         let creds = self.credentials_provider.get_credentials().await;
 
-        let session_config = self.session_config.clone();
-        let cache = self.spotifyd_state.cache.clone();
-
-        Session::connect(session_config, creds, cache, true)
-            .await
-            .map(|(session, _creds)| session)
+        // TODO: expose is_group
+        Spirc::new(
+            ConnectConfig {
+                name: self.spotifyd_state.device_name.clone(),
+                device_type: self.device_type,
+                is_group: false,
+                initial_volume: self.initial_volume,
+                has_volume_ctrl: self.has_volume_ctrl,
```

Comment:
Just out of curiosity, why is this option exposed? Can a user of `spotifyd` adjust this setting? And if so, why would you want to remove the volume control?

> In `0.6-dev` that option was dropped while refactoring for the dealer, so it might be good to know when you want/need that option back or not^^;

Reply:
`has_volume_ctrl` isn't directly exposed to the user, but is determined based on the type of mixer we are using. So, unless the user wants to disable volume control by setting `volume_control = "none"` (which sets the mixer to an always `100%` mixer), this will be true.

So I'm not sure, whether librespot has some other way of informing Spotify of the type of volume control that is being used, but otherwise I think it would be nice to keep this option around.

Reply:
Interesting use case, but I can now understand why you would use that option. With the dealer there is a `disable_volume` flag instead of `has_volume_ctrl`, but it seems to accomplish the same thing and disables the volume control.

Beside that, there is a new option named `volume_steps`. Tho I don't think that accomplishes what you are looking for if I'm correct?